sycl/include/sycl/platform.hpp

```diff
@@ -9,14 +9,11 @@
 #pragma once
 
 #include <sycl/aspects.hpp>
-#include <sycl/context.hpp>
 #include <sycl/detail/backend_traits.hpp>
-#include <sycl/detail/common.hpp>
 #include <sycl/detail/export.hpp>
 #include <sycl/detail/info_desc_helpers.hpp>
 #include <sycl/detail/owner_less_base.hpp>
 #include <sycl/device_selector.hpp>
-#include <sycl/ext/oneapi/weak_object_base.hpp>
```

Comment:
Is this not used or include via some other header here? I'd hate if we actually need it but get it from an earlier header include from `<sycl/sycl.hpp>`.

Reply:
I see no usage of `weak_object_base` in `platform.hpp` header. 
It would be surprising if core SYCL class API would require oneAPI extension(!).

Reply:
Looks like that's right. `owner_less_base` is the related dependency. Note that it is also related to an extension, but it is an extension that affects `platform` so it makes sense to have the relationship.